packages/variable-resolver/src/browser/env-variable-contribution.ts

```diff
@@ -0,0 +1,36 @@
+/********************************************************************************
+ * Copyright (C) 2019 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from 'inversify';
+import { VariableContribution, VariableRegistry } from './variable';
+import { EnvVariablesServer } from '@theia/core/lib/common/env-variables';
+
+@injectable()
+export class EnvVariableContribution implements VariableContribution {
+
+    @inject(EnvVariablesServer)
+    protected readonly env: EnvVariablesServer;
+
+    async registerVariables(variables: VariableRegistry): Promise<void> {
```

Comment:
Do you think we should document this public API?

Reply:
It is part of the `VariableContribution` interface.

Reply:
It's documented in the Variable Resolver extension's [README](https://github.com/theia-ide/theia/blob/master/packages/variable-resolver/README.md) file

Reply:
@marechal-p thanks, forgot it's simply implementing the method.
It looks like the interface method is undocumented however :(

https://github.com/theia-ide/theia/blob/f9ff23786ed48f611aa335527c78db70427edc54/packages/variable-resolver/src/browser/variable.ts#L49